site/content/docs/cli/constructs-reference.md

```diff
@@ -301,6 +301,87 @@ new MultiStepCheck('multistep-check-1', {
 - `code`: an object with either an `entrypoint` property that points to `.spec.js|ts` file, or a `content` property with
 raw JavaScript / TypeScript as a string.
 
+
+## `TcpCheck`
+
+TCP Checks are ideal for monitoring services and protocols that use TCP, such as FTP, messaging protocols like MQTT and XMPP, among others. The example below shows the following:
+
+- It defines the basic Check properties like `name`, `activated` etc.
+- It defines the `hostname`, `port`, and `data`
+- It defines an array of assertions using the `TcpAssertionBuilder` to assert that:
+  - the total response time is within the given limit
+  - the response contains provided value
+
+
+The file hierarchy looks as follows:
+
+```
+├── __checks__
+│   ├── hello-tcp.check.ts
+```
+
+```ts {title="hello-tcp.check.ts"}
+import { TcpCheck, TcpAssertionBuilder } from 'checkly/constructs'
+
+new TcpCheck('hello-tcp-1', {
+  name: 'Hello TCP',
+  activated: true,
+  maxResponseTime: 10000,
+  degradedResponseTime: 5000,
+  request: {
+    hostname: 'tcpbin.com',
+    port: 4242,
+    data: 'ping\n',
+    ipFamily: 'IPv6',
+    assertions: [
+        TcpAssertionBuilder.responseTime().lessThan(1000),
+        TcpAssertionBuilder.responseData().contains('ping')
+    ]
+  }
+})
+```
+- `maxResponseTime`: The response time in milliseconds where a check should be considered failing.
+- `degradedResponseTime`: The response time in milliseconds where a check should be considered degraded.
+- `request`: An object of the `TcpRequest` type. See the [`TcpRequest` reference](#tcprequest).
+- `shouldFail`: When set to `true`, the check is considered successful if the connection attempt fails. If not specified, the default value is `false`.
+
+> [!NOTE]
+> Failing assertions will cause the check to fail, regardless of the `shouldFail` value
+
+
+
+### `TcpRequest`
+
+The `request` object is a mandatory part of an TCP check.
+
+- `hostname`: The hostname the connection should be made to. Do not include a scheme or a port in the hostname.
+- `port`: The port the connection should be made to.
+- `ipFamily`: The IP family to use for the connection. Defaults to `IPv4`
+- `data`: The data to send to the target host.
```

Comment:
Does it make sense to specify the format here? E.g. "only plain text is accepted"